### PR TITLE
Add the scanpy tool `tl` modules to API agent, next and closing steps. 

### PR DESCRIPTION
**Pull request by bastienchassagnol**

### CODE REVIEW

**Comment by slobentanzer** [comment]
could `scanpy` be a dev group dependency (only installed when installing biochatter locally from the repo)? I am not sure we even want it as an optional dependency from pip. Can something be dev and optional at the same time?

File: test/test_pydantic_class_generation.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import scanpy as sc
+
+from biochatter.api_agent.generate_pydantic_classes_from_module import generate_pydantic_classes
+
+
+def test_generate_pydantic_classes() -> None:
```
Comment (bastienchassagnol) [inline test/test_pydantic_class_generation.py:6]: @slobentanzer is typing relevant for unit functional tests?

Comment (slobentanzer) [reply test/test_pydantic_class_generation.py:6]: no, this can be deactivated

**Review by slobentanzer — CHANGES_REQUESTED** [review]
Nice work! Todos:
- add docparse library
- move scanpy dependency to dev

**Review by bastienchassagnol — COMMENTED** [review]
added the docparse-string python module + set aside scanpy into the optional dependencies

**Review by slobentanzer — APPROVED** [review]
Great work @bastienchassagnol @mengerj @vd-dragan21 @kvitoslava-yarish :)

Merging

**Pull request merged**